### ISSUE TITLE
FP Implementation

### ISSUE THREAD

**Issue by patrick1011** [dgilland/pydash#100]
Hi Folks,

Great library, been using it over the last month or two and enjoying it.  I was just wondering if anyone has ever looked into porting over FP?  I quite like wrapping my functions in `flow` which really necessitates 'auto-curried iteratee-first data-last' functions to quote the FP blurb.  I've taken a look at how FP is written and would be up for implementing something similar in pydash if you think it's a sensible idea!  If so I'll get something together a make a PR!

Cheers,

Patrick

**Comment by dgilland** [comment]
I'm not necessarily opposed to the idea in theory, but I'd like to hear what kind of design you would propose to port `lodash/fp` since it seems like it would require a large amount of work to do (at least from an initial glance at https://github.com/lodash/lodash/wiki/FP-Guide).

**Comment by patrick1011** [comment]
Have you seen https://github.com/lodash/lodash/tree/4.17.4-npm/fp ?  https://github.com/lodash/lodash/blob/4.17.4-npm/fp/_baseConvert.js is where most of the action takes place.  

I think the base conversion would generally be a variation on the below and could cover upwards of half of the library.
```
def pydash_fp_func(collection):
    def wrapped(predicate):
        pydash_func(collection, predicate)
    return wrapped
```
Special cases would of course likely crop up.  And the tests would need to be ported over too.

**Comment by dgilland** [comment]
Ok, so basically there would be a set of wrapper functions to convert pydash functions to fp-style functions. Sounds reasonable enough. Let's keep the changes in a feature branch like `feature/fp`.

**Comment by patrick1011** [comment]
Yeah.  Nice one - I'll get something together for you to take a look at.

**Comment by patrick1011** [comment]
Just to let you know I'm still working on this - progress has been slow because I've been ill the last week.

**Comment by dgilland** [comment]
Any progress on this?

**Comment by impguard** [comment]
:(

Unfortunate. It makes a lot more sense as a builtin for the library (from one perspective), since both would always be in sync. Instead of having one library trail the other.

Though I can see why the decision was made.
